src/objective-c/tests/BUILD

```diff
@@ -341,6 +341,7 @@ grpc_objc_testing_library(
         "//src/core:cf_event_engine",
         "//test/core/event_engine/test_suite/posix:oracle_event_engine_posix",
         "//test/core/event_engine/test_suite/tests:client",
+        "//test/core/event_engine/test_suite/tests:dns",
```

Comment:
I'm not sure the DNS tests will be able to run on iOS, they are going to rely on a twisted-based Python server running on the same platform. There may be no harm in including this for now, but likely the DNS tests will be disabled for this implementation.

Reply:
I added #ifndef of `GRPC_IOS_EVENT_ENGINE_CLIENT` around to disable the new tests that depending the custom server as well.
The tests moved from `test/core/iomgr/resolve_address_test.cc` are passing.

Reply:
Glad to know the twisted server works in the test environment. We can work with the ifdefs for now.

Unfortunately, because you force-pushed to the branch, I can't see if the DNS tests are actually running in the CI environment because the results were cached from a previous run that I don't know how to find now. The links disappear when you rewrite history.

Reply:
Oh I didn't know it. I only rewrote the address comments one, though the commit history would be cleaner.

Reply:
I found it in my browser history:
https://source.cloud.google.com/results/invocations/474fcf95-0375-43e1-8825-cec3b6b80c24/targets

PS, the new tests in `test/core/event_engine/cf/cf_engine_test.cc` are only run on Mac for now in CI.
I have manually tested them on iOS locally, I'll need sometime to restructure the test code before I can submit it (a new commit or separate PR)

Reply:
For the `the twisted server`, it doesn't work on iOS or Mac, the tests depend on it didn't run.

Reply:
I see. Looks like the tests are indeed running, that's great.